.gitlab-ci.yml

```diff
@@ -52,104 +54,103 @@ benchmark container:
     - docker build --build-arg CI_COMMIT_SHA=$CI_COMMIT_SHA -t registry.gitlab.com/rirvm/rir_mirror/benchmark:$CI_COMMIT_SHA .
     - docker push registry.gitlab.com/rirvm/rir_mirror/benchmark:$CI_COMMIT_SHA
 
-tests_release:
-  image: registry.gitlab.com/rirvm/rir_mirror:$CI_COMMIT_SHA
-  variables:
-      GIT_STRATEGY: none
-  stage: Run tests
-  script:
-    - cd /opt/rir/build/release
-    - bin/gnur-make-tests test-all-devel
-    - ../../tools/check-gnur-make-tests-error
-    - bin/gnur-make-tests check-recommended
-    - ../../tools/check-gnur-make-tests-error
-
-test_types_and_serialize:
+# Run check-all in release mode  (additionally check pir types)
+test_release:
   image: registry.gitlab.com/rirvm/rir_mirror:$CI_COMMIT_SHA
   variables:
-      GIT_STRATEGY: none
+    GIT_STRATEGY: none
   stage: Run tests
+  retry: 1
   script:
     - cd /opt/rir/build/release
     - RIR_CHECK_PIR_TYPES=1 bin/tests
     - RIR_CHECK_PIR_TYPES=1 bin/gnur-make-tests check-all
     - ../../tools/check-gnur-make-tests-error
-#    - RIR_SERIALIZE_CHAOS=1 bin/tests
-#    - RIR_SERIALIZE_CHAOS=1 bin/gnur-make-tests test-all-basics
-#    - RIR_SERIALIZE_CHAOS=3 bin/tests
-
-tests_fullverifier:
-  image: registry.gitlab.com/rirvm/rir_mirror:$CI_COMMIT_SHA
-  variables:
-      GIT_STRATEGY: none
-  stage: Run tests
-  script:
-    - cd /opt/rir/build/fullverifier
-    - bin/tests
-    - bin/gnur-make-tests check
-    - PIR_WARMUP=4 ./bin/gnur-make-tests check
-    - PIR_ENABLE=force ./bin/gnur-make-tests check
 
+# Run some tests in the debug mode
 tests_debug:
   image: registry.gitlab.com/rirvm/rir_mirror:$CI_COMMIT_SHA
   variables:
-      GIT_STRATEGY: none
+    GIT_STRATEGY: none
   stage: Run tests
+  retry: 1
   script:
     - cd /opt/rir/build/debugopt
     - export UNSOUND_OPTS=off
-    - bin/tests
     - R_ENABLE_JIT=0 ./bin/tests
     - R_ENABLE_JIT=1 ./bin/tests
     - R_ENABLE_JIT=2 ./bin/tests
     - R_ENABLE_JIT=3 ./bin/tests
     - PIR_ENABLE=off ./bin/tests
     - PIR_ENABLE=force ./bin/tests
-    - for i in `seq 1 5`; do PIR_DEOPT_CHAOS_SEED=$i PIR_DEOPT_CHAOS=1 ./bin/tests ; done
     - ./bin/gnur-make-tests check-devel
     - ../../tools/check-gnur-make-tests-error
 
-test_sanitize:
+# Run some tests in the full verify (release) mode
+tests_debug:
```

Comment:
Should this configuration be `tests_fullverifier`?

Reply:
thanks!